clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1351,7 +1352,7 @@ class SyclKernelFieldChecker : public SyclKernelFieldHandler {
   void checkBufferLocationType(QualType PropTy, SourceLocation Loc) {
     const auto *PropDecl =
         cast<ClassTemplateSpecializationDecl>(PropTy->getAsRecordDecl());
-    if (PropDecl->getTemplateArgs().size() != 1) {
+    if (PropDecl != nullptr && PropDecl->getTemplateArgs().size() != 1) {
```

Comment:
We can make it simpler
```suggestion
    if (PropDecl && PropDecl->getTemplateArgs().size() != 1) {
```